pkg/connector/account.go

```diff
@@ -276,13 +286,108 @@ func (o *accountResourceType) Grant(ctx context.Context, principal *v2.Resource,
 		TargetType:       awsSsoAdminTypes.TargetTypeAwsAccount,
 	}
 
-	if _, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp); err != nil {
+	createOut, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp)
+	if err != nil {
 		return nil, err
 	}
 
+	attemptCount := 1
+	attemptLogger := l.With(
+		zap.Int("attempt", attemptCount),
+		zap.String("request_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.RequestId)),
+		zap.String("principal_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PrincipalId)),
+		zap.String("principal_type", string(createOut.AccountAssignmentCreationStatus.PrincipalType)),
+		zap.String("permission_set_arn", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PermissionSetArn)),
+	)
+
+	complete, err := o.checkCreateAccountAssignmentStatus(ctx, attemptLogger, createOut.AccountAssignmentCreationStatus)
+	if err != nil {
+		var ae *awsSsoAdminTypes.AccessDeniedException
+		if errors.As(err, &ae) {
+			l.Info("aws-connector: access denied while attempting to check status. Assuming account assignment creation is complete.", zap.Error(err))
+			complete = true
+		} else {
+			return nil, err
+		}
```

Comment:
I don't think we should assume it worked here. But we also don't want to have had it worked and we report that it didn't. Hmm. Maybe this is fine.